documentation/IDTA-01001/modules/ROOT/pages/includes/IDTA-01001_ValueOnly.adoc

```diff
@@ -11,46 +11,329 @@ Plattform Industrie 4.0; Anna Salari, Publik. Agentur für Kommunikation GmbH, d
+2+|*Identifiables*
+|AssetAdministrationShell a|assetInformation, submodels
+|Submodel a| (submodelElements)
+
+====
+Qualifiers are ignored. This also holds for qualifiers of kind "ValueQualifier".
+====
+
+|*SubmodelElements* a| 
+
+====
+Qualifiers are ignored. This also holds for qualifiers of kind "ValueQualifier".
+====
+
+|SubmodelElementCollection a| (value) 
+
+====
+If the collection contains submodel elements of type
+ Operation or Capability: these are ignored in the Value-Only serialization
+====
+
+|SubmodelElementList a| (value) 
+
+if defined: type equal to SubmodelElementList/valueTypeListElement
+
+====
+If the submodel element list is part the Value-Only serialization of its parent element (e.g. in the context of a submodel or submodel element collection) and the list consists of submodel elements of type Operation or Capability the complete list is ignored. 
+
+If the Value-Only serialization of the list is explicitly requested without context a corresponding error message shall be returned. 
+====
+
+|Entity |statements, globalAssetId, specificAssetId
+|BasicEventElement a|observed
+|Capability a| -
+|Operation a| -
+
+====
+There is not Value-Only serialization for an Operation. However, a Value-Only serialization is useful for its arguments when executing the operation (see Value-Only serialization of "OperationVariable")
+====
+
+|RelationshipElement a|first, second
+|AnnotatedRelationshipElement a|first, second, annotations
+
+2+|*DataElements*
+|Property a| (value)
+
+type equal to Property/valueType
+
+====
+valueId is ignored in Value-Only serialization
+====
+|MultilanguageProperty a| (value)
+
+====
+valueId is ignored in Value-Only serialization
+====
+|Range a|min, max
+
+type equal to Range/valueType
+
+====
+In the Value-Only serialization only number ranges are supported
+====
+
+|ReferenceElement a|(value)
+
+|Blob a|value, contentType
+|File a|value, contentType
+
+2+|*Data Types*
+|OperationVariable a| (value)
+
+|MultiLanguageTextType a| The value of this type is identicial to its "Normal"-format serialization: it consists of a list of pairs: a language tag and the value in the corresponding language
+
+[listing]
+....
+<language-tag> ::= 
+	language tag as defined for values of type langString, i.e.
+	in accordance with IETF https://tools.ietf.org/html/rfc5646#page-5[RFC 5646], 
+	the language names match the following regular expression:
+
+		^[a-z]\{2,4}(-[A-Z][a-z]\{3})?(-([A-Z]\{2}|[0-9]\{3}))?$
+	
+....	
+
+	e.g. "en" for Englisch
+|===
 
-* Submodel elements defined in the submodel other than the ones mentioned above are not subject to serialization of that SerializationModifier.
 
-*Data type to value mapping* footnote:[cf. https://openmanufacturingplatform.github.io/sds-documentation/bamm-specification/2.0.0/datatypes.html]
+A submodel element is omitted if no value is defined for it for the requesting submodel or submodel element.
```

Comment:
I do expect "tons of empty elements" for optional elements. This is why they are optional. As far as I remember there was a decision to remove them from the payload. And the sentence shall reflect this decision.l

Reply:
It seems like how to handle empty elements is one of the most important question. This especially becomes important when thinking about serialization of SubmodelElementLists.

What is the valueOnly serialization of a SubmodelElementList containing 2 properties, first one is empty, second one is not. Omitting empty values this would result in 
```JSON
[
   "some value"
]
```

and it would be impossible to tell if `some value` is the value of the first or second element in the list.

Omitting every empty element could lead to somewhat strange result, e.g. a submodel containing multiple collections, each only containing elements with empty value would just be serialized to an completely empty string following this logic. Not sure if that is what I would expect.

Furthermore, I do not have the feeling that having "tons of empty elements" would be a realistic problem because there are also parameters like level which allows to limit the depth of the returned structure. 

Not sure what is the best solution here but just enforcing all empty elements to be omitted obviously causes problems.

Reply:
I added issue https://github.com/admin-shell-io/aas-specs/issues/398 to request 

`For of Property within a Submodel with kind=Instance Property/value shall be defined.`

`For of MultiLanguageProperty within a Submodel with kind=Instance MultiLanguageProperty/value shall be defined.`

Reply:
I agree that there might be some misunderstanding regarding the term *optional* because it may refer to the meta-model or to JSON. My primary issue is that we agree on a JSON serialization and once we have that we can write the rules how to generate it.

Regarding Property valueOnly serialization, what is the correct/expected JSON output for serializing a submodel (idShort *submodel*) that contains a SubmodelElementCollection (idShort *collection*) which again contains a single Property with idShort *property* and no value?

A simplified version of regular JSON serialization could look like this
```JSON
{
	"modelType": "Submodel",
	"idShort": "submodel",	
	"submodelElements": [
		{
			"modelType": "SubmodelElementCollection"
			"idShort": "collection",
			"value": [
				{
					"modelType": "Property",
					"idShort": "property",
					"valueType": "xs:string"					
				}
			]
		}
	]	
}
```

I can think of the following possible results

1. 
```JSON
{
    "collection": {
        "property": null
    }
}
```
2. 
```JSON
{
    "collection" : { }
}
```
3. 
```JSON
{ }
```
4. 
```JSON
null
```

At first glance, only the first option looks viable to me because of the issue with SumodelElementLists as discussed in an earlier comment.
Using this to deduct serialization rules from it, this would imply that all values have to be serialized at all times resp. `null` to be used to indicate a value is not present.